### PR TITLE
Restore student repo edit button

### PR DESCRIPTION
**Pull request by webecke**
## Overview
Somehow, somewhere, the repo edit button got removed from the student view. This puts it back.

## Details
- Put button back
- Reload page upon successful repo change

## Testing

- [ ] Added/updated unit tests
- [ ] Tested edge cases
- [X] Manual testing (if needed)

### CODE REVIEW

File: src/main/resources/frontend/src/App.vue [hunk 1]
```diff
@@ -36,11 +36,12 @@ onMounted(async () => {
   await useAppConfigStore().updateConfig();
 });
 
-const openRepoEditor = reactive({ value: false });
+const openRepoEditor = ref<boolean>(false);
 
 const repoEditDone = () => {
   openRepoEditor.value = false;
   loadUser();
+  location.reload();
```
Comment (frozenfrank) [inline src/main/resources/frontend/src/App.vue:44]: Does it have to reload the entire page, or is there a more minimal refresh that can be applied?

File: src/main/resources/frontend/src/App.vue [hunk 2]
```diff
@@ -49,16 +50,17 @@ const repoEditDone = () => {
...
   <main>
-    <PopUp
-      id="repoEditorPopUp"
-      v-if="openRepoEditor.value"
-      @closePopUp="openRepoEditor.value = false"
-    >
+    <PopUp id="repoEditorPopUp" v-if="openRepoEditor" @closePopUp="openRepoEditor = false">
+      <h2>Change Repo</h2>
       <RepoEditor @repoEditSuccess="repoEditDone" :user="useAuthStore().user" />
+      This will not affect previous submissions.
```
Comment (frozenfrank) [inline src/main/resources/frontend/src/App.vue:63]: I like this clarification.

File: src/main/resources/frontend/src/App.vue [hunk 3]
```diff
@@ -49,16 +50,17 @@ const repoEditDone = () => {
     <h1>CS 240 Autograder</h1>
     <h3>This is where you can submit your assignments and view your scores.</h3>
     <p>{{ greeting }} <a v-if="useAuthStore().isLoggedIn" @click="logOut">Logout</a></p>
-    <p>{{ useAuthStore().user?.repoUrl }}</p>
+    <p @click="openRepoEditor = true" style="cursor: pointer">
+      {{ useAuthStore().user?.repoUrl }}
+      <i class="fa-solid fa-pen-to-square" v-if="useAuthStore().user?.repoUrl" />
```
Comment (frozenfrank) [inline src/main/resources/frontend/src/App.vue:55]: This `v-if="..."` statement, should it be placed on the `<p>` element on line 53? If the value is missing, what would `useAuthStore().user?.repoUrl` display anyways? That should be guarded by the `v-if`; a `v-else` could be used if you want to carefully define the behavior when no URL already exists.

File: src/main/resources/frontend/src/App.vue [hunk 4]
```diff
@@ -36,11 +36,12 @@ onMounted(async () => {
   await useAppConfigStore().updateConfig();
 });
 
-const openRepoEditor = reactive({ value: false });
+const openRepoEditor = ref<boolean>(false);
```
Comment (frozenfrank) [inline src/main/resources/frontend/src/App.vue:39]: This looks like a good change that better conforms to the patterns and practices of VUE.

**Review by frozenfrank — COMMENTED** [review]
Looks pretty good, but I did leave a few technical comments that may be worth considering.

Thanks for bringing this to pass!

File: src/main/resources/frontend/src/App.vue (hunk 3 above)
Comment (webecke) [reply src/main/resources/frontend/src/App.vue:55]: Good point

**Review by 19mdavenport — APPROVED** [review]
